lalrpop-test/src/arg_pattern.lalrpop

```diff
@@ -0,0 +1,19 @@
+grammar;
+
+Barm<A, B> = { A B, B A }
+DeepBar<A, B, C> = {
+    A Barm<B, C>,
+}
+Bar: (String, String) = {
+    <a:r"[c-z]"> <b:r"[c-z]"> => (a.to_string(), b.to_string())
+}
+
+pub Foo: String = {
+    <(a, b):Barm<"a", "b">> => {
+        a.to_string() + b
+    },
+    <(a, (b, c)):DeepBar<"A", "B", "C">> => {
+        a.to_string() + b + c
+    },
+    <(a,b):Bar> => a + &b
```

Comment:
One super simple addition to add is a `mut` somewhere here. Just as a sanity check